commons/src/main/java/org/infinispan/commons/equivalence/EntryEquivalence.java

```diff
@@ -0,0 +1,62 @@
+package org.infinispan.commons.equivalence;
+
+import java.util.Map;
+
+/**
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class EntryEquivalence<K, V> implements Equivalence<Map.Entry<K, V>> {
```

Comment:
I'm not sure it's needed there either, AFAICT all our other `retainAll()` implementations assume that `c.contains(e)` uses the same Equivalence and don't try to copy the input in another set with the cache's Equivalence.